site/src/modules/resources/AppLink/BaseIcon.tsx

```diff
@@ -4,14 +4,18 @@ import type { FC } from "react";
 
 interface BaseIconProps {
 	app: WorkspaceApp;
+	onError?: () => void;
 }
 
-export const BaseIcon: FC<BaseIconProps> = ({ app }) => {
+export const BaseIcon: FC<BaseIconProps> = ({ app, onError }) => {
 	return app.icon ? (
 		<img
 			alt={`${app.display_name} Icon`}
 			src={app.icon}
 			style={{ pointerEvents: "none" }}
+			onError={() => {
+				onError?.();
+			}}
```

Comment:
The naming here is kind of confusing IMO. Can we rename the `onError` function variable so it's distinct?

Reply:
Good call, renamed to `onIconPathError`